.circleci/config.yml

```diff
@@ -0,0 +1,89 @@
+version: 2
+jobs:
+  test-build:
+    machine:
+      docker_layer_caching: true
+    steps:
+      - checkout
+      - run: VERSION="v2.6.2" make build
+
+  build:
+    machine:
+      docker_layer_caching: true
+    steps:
+      - checkout
+      - run: VERSION="${CIRCLE_TAG}" make build
+
+      - run:
+          name: List available images
+          command: docker images fundocker/learninglocker
+
+      - run:
+          name: Store docker image in cache
+          command: |
+            docker save \
+              -o docker/images/learninglocker.tar \
+              fundocker/learninglocker:${CIRCLE_TAG}
+
+      - save_cache:
+          paths:
+            - ~/docker/images/
+          key: learninglocker-image-{{ .Revision }}
+
+  hub:
+    machine: true
+    steps:
+      - restore_cache:
+          keys:
+            - learninglocker-image-{{ .Revision }}
+
+      - run:
+          name: Load images to docker engine
+          command: |
+            docker load < docker/images/learninglocker.tar
+      # Login to DockerHub to Publish new images
+      #
+      # Nota bene: you'll need to define the following secrets environment vars
+      # in CircleCI interface:
+      #
+      #   - DOCKER_USER
+      #   - DOCKER_PASS
+      - run:
+          name: Login to DockerHub
+          command: echo "$DOCKER_PASS" | docker login -u "$DOCKER_USER" --password-stdin
+
+      - run:
+          name: List available images
+          command: docker images fundocker/learninglocker
+
+      - run:
+          name: Publish images
+          command: |
+            docker push fundocker/learninglocker:${CIRCLE_TAG}
+
+workflows:
+  version: 2
+  learninglocker:
+    jobs:
+      - test-build:
+          filters:
+            tags:
+              ignore: /.*/
+            branches:
+              only: /.*/
+
+      - build:
+          filters:
+            tags:
+              only: /^v.*$/
```

Comment:
I would always trigger the build. Only the hub step should be restricted be a tag following this pattern.